ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -108,10 +112,57 @@ prop_Mempool_InvalidTxsNeverAdded bc txs =
     genTxIsInvalid (TestGenTx (InvalidTestTx _)) = True
     genTxIsInvalid _                             = False
 
+-- | Test that all valid transactions added to a 'Mempool' via 'addTxs' are
+-- appropriately represented in the trace of events.
+prop_Mempool_TraceValidTxs :: Chain TestBlock
+                           -> [TestTx]
+                           -> Property
+prop_Mempool_TraceValidTxs bc txs =
+    testAddTxsWithTrace bc txs traceProp
+  where
+    traceProp :: [GenTx TestBlock] -> [TraceEventMempool TestBlock] -> Property
+    traceProp genTxs es = case (filter isAddTxsEvent es) of
+      [TraceMempoolAddTxs addedTxs _] ->
+        sortGenTxs (filter genTxIsValid genTxs) === sortGenTxs addedTxs
+      _ -> es === [] .&&. (filter genTxIsValid genTxs) === []
+    --
+    genTxIsValid :: GenTx TestBlock -> Bool
+    genTxIsValid (TestGenTx (ValidTestTx _)) = True
+    genTxIsValid _                           = False
+    --
+    isAddTxsEvent :: TraceEventMempool blk -> Bool
+    isAddTxsEvent (TraceMempoolAddTxs _ _) = True
+    isAddTxsEvent _                        = False
+
+-- | Test that all invalid rejected transactions returned from 'addTxs' are
+-- appropriately represented in the trace of events.
+prop_Mempool_TraceRejectedTxs :: Chain TestBlock
```

Comment:
And now we still need a property for `TraceRemovedTxs` :grin: (I know, first we need to extend the test blocks)

Reply:
I'll add a TODO comment and raise an issue ^

Reply:
[Added a TODO comment.](https://github.com/input-output-hk/ouroboros-network/pull/696/files#diff-c2e021fd5c568658ad5f3838f11fbf6aR163)